medcat/cat.py

```diff
@@ -454,9 +442,9 @@ def _print_stats(self,
                 if use_cui_doc_limit:
                     _cuis = set([ann['cui'] for ann in anns])
                     if _cuis:
-                        filters['cuis'] = intersect_nonempty_set(_cuis, extra_cui_filter)
+                        local_filters.cuis = intersect_nonempty_set(_cuis, extra_cui_filter)
```

Comment:
Must be an intersection only, not union.  

Agree documentation on the usage of all the possible filters in necessary. 

Also a warning that there are additional filters maybe needed. It's easy to forget about the model filters.

Reply:
okay - can you add a task or an example doc string that @mart-r can add. It must be an intersection due to my thinking above right?

Reply:
Exactly.

An empty Model filter set assumes that all concepts present in the model and are within the Model filter. This is the Superset.

To clarify using set notation, the order of filters must follow this logic:
extra_cui_filter ⊆ MCT filter ⊆ Model filter

Although there is nothing stopping you from adding concepts to the extra filter that are not present in the model. This is pointless.

Reply:
cool - got it - can we add that in the config.linking.filters documentation - or a task on clickup to address later

Reply:
I think this should be documented in the `train_supervised` method since that is the only one where all 3 are present.

But documenting this under the filters as well does make sense, too.

I think I could add something like this for linking.filters documentation:
```
    """These describe the linking filters used alongside the model.

    When no CUIs nor exlcuded CUIs are specified (the sets are empty),
    all CUIs are accepted.
    If there are CUIs specified then only those will be accepted.
    If there are excluded CUIs specified, they are excluded.

    In some cases, there are extra filters as well as MedCATtrainer (MCT) export filters.
    These are expcted to follow the following:
    extra_cui_filter ⊆ MCT filter ⊆ Model/config filter

    While any other CUIs can be included in the the extra CUI filter or the MCT filter,
    they would not have any real effect.
    """
```
And for the `train_supervised` method I could add
```
        When filtering, the filters within the CAT model are used first,
        then the ones from MedCATtrainer (MCT) export filters,
        and finally the extra_cui_filter (if set).
        That is to say, the expectation is:
        extra_cui_filter ⊆ MCT filter ⊆ Model filter.
```

Does that track or did I misunderstand something?

Reply:
Perfect. Just keep the name: "Model/config filter" for both.